docs/gateway.md

```diff
@@ -0,0 +1,70 @@
+# Gateway for Tekton triggers
+
+Leveraging kcp, workload clusters are an infrastructure that end users may not be aware of and not allowed to access directly.
+With the approach described for [phase 1](../README.md#phase-1) controllers are directly running on these workload clusters.
+
+This introduces a challenge. For trigger integration, external systems or users need to send requests to the listeners on the workload cluster.
+The approach taken for addressing it is to introduce a gateway (an haproxy container) that forwards the requests to listeners part of of the workload cluster infrastructure.
+
+## Architecture
+
+![Gateway architecture](../images/haproxy.png)
+
+- GLB is kcp global load balancer. This is work in progress. Currently a development version is based on an ingress-controller that automatically configures an envoy proxy.
+- Ingress, Service, Deployment and ConfigMap for the HAProxy based gateway are created in kcp, which syncs them onto the workload clusters
+
+The request flow is as follows:
+1. an external system, GitHub for instance, triggers an http call
+2. the GLB forwards the call to the ingress router running on a workload cluster
+3. the ingress router forwards the call to its backend the HAProxy container
+4. HAProxy uses path based routing to forward `/trigger` requests to the trigger Service in the trigger namespace
+5. The Service, implemented through iptables rules most of the times forward the packets to the trigger pod
+6. The trigger pod processes the request
+
+## Installation
+
+having kubectl pointing to your kcp organisation
+
+~~~
+kubectl kcp workspace create infra --enter
+kubectl create -f ./gitops/triggers/gateway/haproxy-cfg-cm.yaml
+kubectl create -f ./gitops/triggers/gateway/haproxy-deployment.yaml
+kubectl create -f ./gitops/triggers/gateway/haproxy-service.yaml
+kubectl create -f ./gitops/triggers/gateway/haproxy-ingress.yaml
+~~~
+
+HAProxy configuration can be amended through the ConfigMap. See the section below.
+
+## Configuration
+
+Connection settings, support for https can be configured by amending the HAProxy configuration contained in the ConfigMap.
+This will get streamlined with the automation of the installation of Tekton triggers.
+
+The first use case for the gateway is to forward requests to the listeners of Tekton triggers. This may also be used to proxy other services.
+Therefore a frontend can configured for a different path and forward queries to a different backend.
```

Comment:
reword: 'Therefore a frontend can be configured so that different paths are forwarded to different backends.'

Reply:
Thanks.